auction-server/src/auction/service/conclude_auction.rs

```diff
@@ -20,59 +20,54 @@ where
     Service<T>: AuctionManager<T>,
 {
     pub async fn conclude_auction(&self, input: ConcludeAuctionInput<T>) -> anyhow::Result<()> {
-        let auction = input.auction;
+        let mut auction = input.auction;
+        tracing::info!(chain_id = self.config.chain_id, auction_id =? auction.id, permission_key = auction.permission_key.to_string(), "Concluding auction");
         if let Some(tx_hash) = auction.tx_hash.clone() {
             let bids = self
                 .repo
-                .get_in_memory_submitted_bids_for_auction(auction.clone())
+                .get_in_memory_submitted_bids_for_auction(&auction)
                 .await;
 
-            if let Some(bid_statuses) = self
+            let bid_statuses = self
                 .get_bid_results(
                     bids.clone(),
                     entities::BidStatusAuction {
                         id: auction.id,
                         tx_hash,
                     },
                 )
-                .await?
-            {
-                let auction = self
-                    .repo
-                    .conclude_auction(auction)
-                    .await
-                    .map_err(|e| anyhow::anyhow!("Failed to conclude auction: {:?}", e))?;
+                .await?;
 
-                join_all(
-                    bid_statuses
-                        .iter()
-                        .enumerate()
-                        .filter_map(|(index, bid_status)| match bids.get(index) {
-                            Some(bid) => Some(self.update_bid_status(UpdateBidStatusInput {
+            join_all(
+                bid_statuses
+                    .iter()
+                    .zip(bids.iter())
+                    .filter_map(|(status, bid)| {
+                        status.as_ref().map(|status| {
+                            self.update_bid_status(UpdateBidStatusInput {
                                 bid:        bid.clone(),
-                                new_status: bid_status.clone(),
-                            })),
-                            None => {
-                                tracing::error!(
-                                    bids = ?bids,
-                                    bid_statuses = ?bid_statuses,
-                                    auction = ?auction,
-                                    "Bids array is smaller than statuses array",
-                                );
-                                None
-                            }
-                        }),
-                )
-                .await;
+                                new_status: status.clone(),
+                            })
+                        })
+                    }),
+            )
+            .await;
+
 
-                if self
-                    .repo
-                    .get_in_memory_submitted_bids_for_auction(auction.clone())
+            if bid_statuses.iter().all(|status| status.is_some()) {
```

Comment:
I think it's better to have the previous logic here
`self.repo.get_in_memory_submitted_bids_for_auction(auction.clone()).is_empty()`